src/index.ts

```diff
@@ -7,6 +7,38 @@ import { Event, IEvent } from "oni-types"
 
 export type DisposeFunction = () => void
 
+/**
+ * API surface area for our Achievements integration
+ */
+export namespace Achievements {
+
+    /**
+     * Definition of an achievement
+     *
+     * An achievement consists of a name, title, along with one or more sub goals
+     */
+    export interface AchievementDefinition {
+        uniqueId: string
+        name: string
+        description: string
+
+        goals: AchievementGoalDefinition[]
+    }
+
+    export interface AchievementGoalDefinition {
```

Comment:
Since `AchievementDefinition` and `AchievementGoalDefinition` are under `Achievement`, I think that the common prefix can be safely omitted.
Also, both end with `Definition` which is redundant too.
How about `Definition` and `goal` respectively?

Reply:
Ah ya, good catch - I like that better.